### PR TITLE
[analyzer] fix crash on binding to symbolic region with `void *` type

### PR DESCRIPTION
**Pull request by pskrgag**
As reported in https://github.com/llvm/llvm-project/pull/103714#issuecomment-2295769193. CSA crashes on trying to bind value to symbolic region with `void *`. This happens when such region gets passed as inline asm input and engine tries to bind `UnknownVal` to that region.

Fix it by changing type from void to char before calling `GetElementZeroRegion`

### CODE REVIEW

**Comment by pskrgag** [comment]
CC @steakhal @NagyDonat

**Comment by NagyDonat** [comment]
To me this solution seems to be a bit hacky -- I don't like that we need to scatter "handle `void *` as if it was `char *`" special cases in various parts of the analyzer (I vaguely recall that I have also seen similar hacks elsewhere).

I'd prefer solutions that are as generic as possible and ensure that `void *` is consistently handled like `char *` if that's what we want. (By the way, will the analyzer be able to use this `UnknownVal` bound to the `void *` pointer? If not, then just avoid the binding.)

These are not blocking issues, feel free to merge this commit to fix the crash (instead of waiting for a theoretically better solution).

File: clang/test/Analysis/asm.cpp [hunk 1]
```diff
@@ -40,3 +40,12 @@ void testInlineAsmMemcpyUninit(void)
...
 }
+
+void *globalPtr;
+
+void testNoCrash()
+{
+  // Use global pointer to make it symbolic. Then engine will try to bind
+  // value to first element of type void * and should not crash.
+  asm ("" : : "a"(globalPtr)); // no crash
+}
```
Comment (steakhal) [inline clang/test/Analysis/asm.cpp:51]:
```suggestion

void testAsmWithVoidPtrArgument()
{
  extern void *globalVoidPtr;
  clang_analyzer_dump(*(int *)globalVoidPtr); // expected-warning-re {{reg_${{[0-9]+}}<int Element{SymRegion{reg_${{[0-9]+}}<void * globalVoidPtr>},0 S64b,int}>}}
  clang_analyzer_dump_ptr(globalVoidPtr); // expected-warning-re {{&SymRegion{reg_${{[0-9]+}}<void * globalVoidPtr>}}}
  asm ("" : : "a"(globalVoidPtr)); // no crash
  clang_analyzer_dump(*(int *)globalVoidPtr); // expected-warning {{Unknown}}
  clang_analyzer_dump_ptr(globalVoidPtr); // expected-warning-re {{&SymRegion{reg_${{[0-9]+}}<void * globalVoidPtr>}}}
}
```

**Comment by steakhal** [comment]
LGTM, thanks!

**Review by steakhal — APPROVED** [review]

**Comment by steakhal** [comment]
Let's wait for the premerge tests before merging.

**Comment by pskrgag** [comment]
added missing declarations. should fix the CI

**Review by NagyDonat — APPROVED** [review]
Bikeshedding: let's delete that empty line.

Otherwise LGTM.

**Pull request merged**